packages/accounts-controller/src/AccountsController.ts

```diff
@@ -302,7 +302,6 @@ export class AccountsController extends BaseController<
         metadata: { ...account.metadata, name: accountName },
       };
       currentState.internalAccounts.accounts[accountId] =
-        // @ts-expect-error Assigning a complex type `T` to `Draft<T>` causes an excessive type instantiation depth error.
```

Comment:
I have absolutely no idea and I'm scared to find out. Although, it's technically _not_ breaking something now, where it was broken before. Maybe some kind of interaction with the `PermissionController` types mediated by Snaps-related imports? That's the only thing I can think of.

Reply:
[This upstream bug](https://github.com/MetaMask/utils/issues/168) currently affects `Draft<T>` types and `update()` calls in our codebase. 

There's not much we can do about it on our end other than add (or remove) `@ts-expect-error` annotations whenever the compiler breaks out with an "excessive type instantiation depth" error.

For `update()` calls, returning a new state instead of mutating in place can sometimes help.

Reply:
What are your thoughts on keeping this comment in place and removing the type assertion instead?

Reply:
Whatever TS version and config my VS Code is using is unhappy about this, but removing both the cast and the `ts-expect-error` seems to work: [c54671b](https://github.com/MetaMask/core/pull/4171/commits/c54671b548c44a2ffb315b9d53f4947c3c0faefc)